### PR TITLE
META-3127 deleted tasks are not allowed for retry

### PR DESCRIPTION
**Pull request by BurakAltas**
## Change description

> Description here

## Type of change
- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

**Review by DillionTan — COMMENTED** [review]
why did we change the logic?

**Comment by BurakAltas** [comment]
Hey Dillion, because tasks with DELETED status are actually ought not to be retried because we create a delete task after them that cleans up the classification vertex. So, every time we want to retry deleted task, the task will throw error stating classification vertex is not found.

**Comment by DillionTan** [comment]
> Hey Dillion, because tasks with DELETED status are actually ought not to be retried because we create a delete task after them that cleans up the classification vertex. So, every time we want to retry deleted task, the task will throw error stating classification vertex is not found.

can we put this info in code comment as well? above line 37

**Comment by BurakAltas** [comment]
Comment is added

**Review by DillionTan — APPROVED** [review]

File: repository/src/main/java/org/apache/atlas/tasks/AtlasTaskService.java [hunk 1]
```diff
@@ -26,9 +26,16 @@ public class AtlasTaskService implements TaskService {
 
     private final AtlasGraph graph;
 
+    private final List<String> retryNotAllowedStatuses;
+
     @Inject
     public AtlasTaskService(AtlasGraph graph) {
         this.graph = graph;
+        retryNotAllowedStatuses = new ArrayList<>();
+        retryNotAllowedStatuses.add(AtlasTask.Status.PENDING.toString());
```
Comment (nikhilbonte21) [inline repository/src/main/java/org/apache/atlas/tasks/AtlasTaskService.java:35]: Rename
retryNotAllowedStatuses -> retryTaskDenyStatuses

OR 

Let us maintain allow status list instead deny, e.g.
retryTaskAllowedStatuses = FAILED, COMPLETED

**Review by nikhilbonte21 — APPROVED** [review]

**Pull request merged**